.github/workflows/conda.yml

```diff
@@ -78,14 +78,18 @@ jobs:
           # If testing the latest `coiled-runtime`, we need to build a Coiled software environment
           # that can be used when running tests
 
-          # Coiled software environment names can't contain "." characters
+          # Coiled software environment names can't contain "." or uppercase characters
           export PYTHON_VERSION_FORMATTED=$(echo "${{ matrix.python-version }}" | sed 's/\.//g' )
+          export OS_FORMATTED=$(python -c "import os; print(os.environ['RUNNER_OS'].lower())")
+          export REF_NAME_FORMATTED=$(echo "$GITHUB_REF_NAME" | sed 's/\./-/g' )
 
+          export COILED_SOFTWARE_NAME_HEAD=dask-engineering/coiled-runtime-$GITHUB_EVENT_NAME
+          export COILED_SOFTWARE_NAME_TAIL=$GITHUB_RUN_ID-$OS_FORMATTED-py$PYTHON_VERSION_FORMATTED
```

Comment:
Just pulling these out into variables so we only have to change them in one place instead both `if`/`else` branches below